### PR TITLE
circleci: vendor go-redis/redis for testing.

### LINKED ISSUE #448: contrib/go-redis/redis not compiling on CI
```
# gopkg.in/DataDog/dd-trace-go.v1/contrib/go-redis/redis
contrib/go-redis/redis/redis.go:75:11: tc.Client.WrapProcess undefined (type *redis.Client has no field or method WrapProcess)
```

This is an upstream change on `master`, as they work toward v7.
It's not released, so we shouldn't change the code yet.

We can probably vendor go-redis/redis at the latest working version, but that may cause type conflicts for users of the tracer, and also other contrib packages eg: redigo.
If that occurs, we'll need to start using dep or modules.

Thoughts?

### PR DESCRIPTION
**Pull request by cgilmour**
The redis package has new development on their master branch for a v7 release, and there are significant API changes. The current contrib may need a rewrite once that API is stable and released. In the meantime, this pins the version of redis that we test against, while still making sure users can choose their own version.

I'm assuming that any users who had been using master would have hit some of the breaking changes and already decided to pick a tag of their own to keep their bits working.
Committing a vendored version into the tree would override the version selected by those users, which isn't very good at the moment.
Once the redis package has a release or RC of v7, we can decide about different import paths for this package. Until then, I think the correct fix is just within our CI.

Fixes #448.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -78,6 +78,13 @@ jobs:
           mkdir -p $GRPC_DEST
           git clone --branch v1.2.0 https://github.com/grpc/grpc-go $GRPC_DEST
 
+    - run:
+        name: Vendor redis v6.15.3
+        # This step vendors go-redis/redis v6.15.3 inside our redis contrib
+        # to allow running the tests against the correct version of
+        # the redis library.
+        command: |
+          git clone --branch v6.15.3 https://github.com/go-redis/redis contrib/go-redis/redis/vendor/github.com/go-redis/redis
```
Comment (gbbr) [inline .circleci/config.yml]: Can we use the same environment variable as for grpc? I find it makes it more readable.

Comment (gbbr) [inline .circleci/config.yml]: I think we have to rename the contribution package to `contrib/go-redis/redis.v6.15.3` just as we do for grpc, and make a copy of it inside `contrib/go-redis/redis` which supports the latest. The latter part you can skip if you think it's too much work and we can do it as part of another PR. But nevertheless we should do it immediately after.

Comment (gbbr) [reply .circleci/config.yml]: Right, as discussed on Slack: it seems there is no v7 release yet, so there is no point to follow this comment for now. Resolving.

Comment (cgilmour) [reply .circleci/config.yml]: Updated this to be consistent but also not too wide.

File: .circleci/config.yml [hunk 2]
```diff
@@ -72,12 +72,20 @@ jobs:
...
+          git clone --branch v1.2.0
+          https://github.com/grpc/grpc-go
+          contrib/google.golang.org/grpc.v12/vendor/google.golang.org/grpc
 
+    - run:
+        name: Vendor redis v6.15.3
+        # This step vendors go-redis/redis v6.15.3 inside our redis contrib
+        # to allow running the tests against the correct version of
+        # the redis library.
+        command: >
```
Comment (gbbr) [inline .circleci/config.yml]: Nice 👍 I didn't know about `>`
